p2psrv/server.go

```diff
@@ -47,6 +48,18 @@ func New(opts *Options) *Server {
 		discoveredNodes.Set(node.ID, node)
 	}
 
+	// AllowedPeers doesn't connect to other sourced nodes
+	allowedOnlyNodes := newNodeMap()
+	if len(opts.AllowedPeers) > 0 {
```

Comment:
What about having this condition at the start so we don't re-write the `knownNodes` and `discoveredNodes`?
```
allowedOnlyNodes := newNodeMap()
knownNodes := cache.NewPrioCache(5)
discoveredNodes := cache.NewRandCache(128)

if len(opts.AllowedPeers) > 0 {
        for _, connectOnlyNode := range opts.AllowedPeers {
	        knownNodes.Set(connectOnlyNode.ID, connectOnlyNode, 0)
	        discoveredNodes.Set(connectOnlyNode.ID, connectOnlyNode)
	        allowedOnlyNodes.Add(connectOnlyNode)
        }
} else {
	for _, node := range opts.KnownNodes {
		knownNodes.Set(node.ID, node, 0)
		discoveredNodes.Set(node.ID, node)
	}
}
```

Reply:
I can see that if this was a high cycled operation it would have an impact, but I think it might be a bit of a style pref.
Do you strongly feel about this ?

Reply:
No, if you prefer it this way, it's fine :)